src/components/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,108 @@
+import {
+  ChangeEvent,
+  ComponentProps,
+  FocusEvent,
+  MouseEvent,
+  ReactNode,
+  useEffect,
+  useState,
+} from "react"
+import styles from "./SearchInput.module.css"
+
+interface SearchInputProps extends ComponentProps<"input"> {
+  searchIcon?: ReactNode
+  clearIcon?: ReactNode
+  onDebounce?: (searchTerm: string) => void
+  debounceDelay?: number
+  onSearchClick?: (searchTerm: string) => void
+  value?: string
+}
+
+export function SearchInput({
+  name,
+  placeholder,
+  searchIcon,
+  clearIcon,
+  onDebounce,
+  debounceDelay = 300,
+  onSearchClick,
+  value = "",
+  ...inputProps
+}: SearchInputProps) {
+  const [searchTerm, setSearchTerm] = useState(value)
+  const [hideCancelButton, setHideCancelButton] = useState(true)
```

Comment:
nit ⛏️ : avoid naming boolean state as a verb. When first reading through this I thought `hideCancelButton` was a function. `setHideCancelButton` is weird to read. Maybe something like `cancelHidden` and `setCancelHidden`?